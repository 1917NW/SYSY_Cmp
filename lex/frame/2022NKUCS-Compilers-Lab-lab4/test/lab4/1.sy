int a;
/*abc\n

*/
//
int test(int a,int b){
    int m;
    a=3;
    return 1;
}
int n;
int main()
{

    int a;
    a = 0x0000aC;
    int b;
    int d;
    int c=test(a,test(b,d));
   
    {
        int b;
    }
    {
        b=3;
    }
    if(a < 5)
        return 1;
    return 0;
}