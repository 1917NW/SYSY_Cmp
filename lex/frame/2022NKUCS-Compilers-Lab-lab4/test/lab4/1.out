[DEBUG LAB4]: 	INT	int	1
[DEBUG LAB4]: 	ID	a	1	94507028099904	index:0
[DEBUG LAB4]: 	SEMICOLON	;	1
[DEBUG LAB4]: 	INT	int	6
[DEBUG LAB4]: 	ID	test	6	94507028100128	index:1
[DEBUG LAB4]: 	LPAREN	(	6
[DEBUG LAB4]: 	INT	int	6
[DEBUG LAB4]: 	ID	a	6	94507028100400	index:2
[DEBUG LAB4]: 	COMMA	,	6
[DEBUG LAB4]: 	INT	int	6
[DEBUG LAB4]: 	ID	b	6	94507028100448	index:3
[DEBUG LAB4]: 	RPAREN	)	6
[DEBUG LAB4]: 	LBRACE	{	6
[DEBUG LAB4]: 	INT	int	7
[DEBUG LAB4]: 	ID	m	7	94507028100704	index:4
[DEBUG LAB4]: 	SEMICOLON	;	7
[DEBUG LAB4]: 	ID	a	8	94507028100608	index:2
[DEBUG LAB4]: 	ASSIGN	=	8
[DEBUG LAB4]: 	NUM	3
[DEBUG LAB4]: 	SEMICOLON	;	8
[DEBUG LAB4]: 	RETURN	return	9
[DEBUG LAB4]: 	NUM	1
[DEBUG LAB4]: 	SEMICOLON	;	9
[DEBUG LAB4]: 	RBRACE	}	10
[DEBUG LAB4]: 	INT	int	11
[DEBUG LAB4]: 	ID	n	11	94507028100752	index:5
[DEBUG LAB4]: 	SEMICOLON	;	11
[DEBUG LAB4]: 	INT	int	12
[DEBUG LAB4]: 	ID	main	12	94507028100800	index:6
[DEBUG LAB4]: 	LPAREN	(	12
[DEBUG LAB4]: 	RPAREN	)	12
[DEBUG LAB4]: 	LBRACE	{	13
[DEBUG LAB4]: 	INT	int	15
[DEBUG LAB4]: 	ID	a	15	94507028100848	index:7
[DEBUG LAB4]: 	SEMICOLON	;	15
[DEBUG LAB4]: 	ID	a	16	94507028100848	index:7
[DEBUG LAB4]: 	ASSIGN	=	16
[DEBUG LAB4]: 	HNUM	172	16
[DEBUG LAB4]: 	SEMICOLON	;	16
[DEBUG LAB4]: 	INT	int	17
[DEBUG LAB4]: 	ID	b	17	94507028101376	index:8
[DEBUG LAB4]: 	SEMICOLON	;	17
[DEBUG LAB4]: 	INT	int	18
[DEBUG LAB4]: 	ID	d	18	94507028101424	index:9
[DEBUG LAB4]: 	SEMICOLON	;	18
[DEBUG LAB4]: 	INT	int	19
[DEBUG LAB4]: 	ID	c	19	94507028101472	index:10
[DEBUG LAB4]: 	ASSIGN	=	19
[DEBUG LAB4]: 	ID	test	19	94507028101040	index:1
[DEBUG LAB4]: 	LPAREN	(	19
[DEBUG LAB4]: 	ID	a	19	94507028101328	index:7
[DEBUG LAB4]: 	COMMA	,	19
[DEBUG LAB4]: 	ID	test	19	94507028101040	index:1
[DEBUG LAB4]: 	LPAREN	(	19
[DEBUG LAB4]: 	ID	b	19	94507028101376	index:8
[DEBUG LAB4]: 	COMMA	,	19
[DEBUG LAB4]: 	ID	d	19	94507028101424	index:9
[DEBUG LAB4]: 	RPAREN	)	19
[DEBUG LAB4]: 	RPAREN	)	19
[DEBUG LAB4]: 	SEMICOLON	;	19
[DEBUG LAB4]: 	LBRACE	{	21
[DEBUG LAB4]: 	INT	int	22
[DEBUG LAB4]: 	ID	b	22	94507028101520	index:11
[DEBUG LAB4]: 	SEMICOLON	;	22
[DEBUG LAB4]: 	RBRACE	}	23
[DEBUG LAB4]: 	LBRACE	{	24
[DEBUG LAB4]: 	ID	b	25	94507028101376	index:8
[DEBUG LAB4]: 	ASSIGN	=	25
[DEBUG LAB4]: 	NUM	3
[DEBUG LAB4]: 	SEMICOLON	;	25
[DEBUG LAB4]: 	RBRACE	}	26
[DEBUG LAB4]: 	IF	if	27
[DEBUG LAB4]: 	LPAREN	(	27
[DEBUG LAB4]: 	ID	a	27	94507028101328	index:7
[DEBUG LAB4]: 	LESS	<	27
[DEBUG LAB4]: 	NUM	5
[DEBUG LAB4]: 	RPAREN	)	27
[DEBUG LAB4]: 	RETURN	return	28
[DEBUG LAB4]: 	NUM	1
[DEBUG LAB4]: 	SEMICOLON	;	28
[DEBUG LAB4]: 	RETURN	return	29
[DEBUG LAB4]: 	NUM	0
[DEBUG LAB4]: 	SEMICOLON	;	29
[DEBUG LAB4]: 	RBRACE	}	30
